### PR TITLE
[API] Support Minimal Artifact Format

### PR DESCRIPTION
**Pull request by quaark**
https://iguazio.atlassian.net/browse/ML-6659

Support Minimal format for artifacts in `GET` requests. Minimal format includes:
```
metadata
status
kind
project
spec:
  producer
  db_key
  size
  framework
  metrics
```

### CODE REVIEW

File: mlrun/db/httpdb.py [hunk 1]
```diff
@@ -996,7 +997,7 @@ def read_artifact(
         error = f"read artifact {project}/{key}"
         # explicitly set artifacts format to 'full' since old servers may default to 'legacy'
         params = {
-            "format": mlrun.common.formatters.ArtifactFormat.full.value,
+            "format": mlrun.common.formatters.ArtifactFormat.full,
```
Comment (TomerShor) [inline mlrun/db/httpdb.py]: Is this still true?
Can't we just pass `_format`?

File: mlrun/db/httpdb.py [hunk 2]
```diff
@@ -1112,7 +1114,7 @@ def list_artifacts(
             "kind": kind,
             "category": category,
             "tree": tree,
-            "format": mlrun.common.formatters.ArtifactFormat.full.value,
+            "format": mlrun.common.formatters.ArtifactFormat.full,
```
Comment (TomerShor) [inline mlrun/db/httpdb.py]: Same here?
```suggestion
            "format": format_,
```

**Review by TomerShor — CHANGES_REQUESTED** [review]
Nice! Minor comments

File: mlrun/common/formatters/artifact.py [hunk 3]
```diff
@@ -13,9 +13,31 @@
...
+                [
+                    "kind",
+                    "metadata",
+                    "status",
+                    "project",
+                    "spec.producer",
+                    "spec.db_key",
+                    "spec.size",
+                    "spec.framework",
+                    "spec.metrics",
```
Comment (mavdryk) [inline mlrun/common/formatters/artifact.py:40]: please don't forget to add `spec.target_path`

**Review by TomerShor — APPROVED** [review]

**Pull request merged**